### PR TITLE
Add native Brainpool support on Node.js

### PR DESCRIPTION
**Pull request by twiss**
Fixes part of #675. Doesn't add support for ED25519 and X25519 yet because the version of OpenSSL included in the latest Node.js (1.0.2m) doesn't support them yet.

### CODE REVIEW

File: src/crypto/public_key/elliptic/curves.js [hunk 1]
```diff
@@ -98,32 +98,32 @@ const curves = {
     node: false // nodeCurves.ed25519 TODO
   },
   curve25519: {
-    oid: [0x06, 0x08, 0x2B, 0x06, 0x01, 0x04, 0x01, 0x97, 0x55, 0x01, 0x05, 0x01],
+    oid: [0x06, 0x0A, 0x2B, 0x06, 0x01, 0x04, 0x01, 0x97, 0x55, 0x01, 0x05, 0x01],
```
Comment (bartbutler) [inline src/crypto/public_key/elliptic/curves.js]: Was the curve25519 OID really wrong?

Comment (bartbutler) [reply src/crypto/public_key/elliptic/curves.js]: Huh, no kidding. Cool.

**Comment by mahrud** [comment]
At least on Mac and Linux, Node.js itself doesn't include OpenSSL. It hooks into the installed version of it. If you're on Linux, the default OpenSSL version probably only has non-patented curves, but if you compile OpenSSL from source you *should* be able to see them.

File: src/crypto/public_key/elliptic/curves.js (hunk 1 above)
Comment (mahrud) [reply src/crypto/public_key/elliptic/curves.js]: I can't count :100:

**Comment by twiss** [comment]
Hm — I actually tried compiling OpenSSL from source and didn't see them. I probably did something wrong, I'll try again.

**Comment by mahrud** [comment]
Ah I see. Thanks for looking into this! I bet it's an issue with endian-ness.
I wonder how gpg2 supports X25519 now :confused:

**Comment by twiss** [comment]
The OpenSSL C API does support it, although it apparently depends on which interface you use. So GPG probably uses that, and so could Node — but I don't know how they interface with OpenSSL so I'm not sure how much work it would be.

**Pull request merged**